### PR TITLE
CCv0 | Adjust tests to using as much components from main as possible

### PR DESCRIPTION
**Pull request by fidencio**
This will help us with the merges from main to CCv0

### CODE REVIEW

**Comment by fidencio** [comment]
> Let me debug what I've done that broke it.

The logic for using the kernel is completely different from what's on main.  I will get this one fixed on main first, and then backport it to CCv0.

It seems to be the only issue we've found so far, which is weirdly positive.

**Comment by fidencio** [comment]
> The logic for using the kernel is completely different from what's on main. I will get this one fixed on main first, and then backport it to CCv0.

https://github.com/kata-containers/kata-containers/pull/7417

**Comment by fidencio** [comment]
SEV tests passed, and I confirmed with @UnmeshDeodhar that it looks good.
```
09:14:21 INFO: Run tests
09:14:22 1..5
09:16:40 ok 1 [cc][kubernetes][containerd][sev] Test SEV unencrypted container launch success
09:17:02 ok 2 [cc][kubernetes][containerd][sev] Test SEV encrypted container launch failure with INVALID measurement
09:17:07 ok 3 [cc][kubernetes][containerd][sev] Test SEV encrypted container launch success with NO measurement
09:17:09 ok 4 [cc][kubernetes][containerd][sev] Test SEV encrypted container launch success with VALID measurement
09:17:20 ok 5 [cc][kubernetes][containerd][sev] Test SEV-ES encrypted container launch success with VALID measurement
```

However, the job itself failed due to:
```
09:17:42 [cleanup_env.sh:58] INFO: Checks that pods were not left
09:17:42 88179b6a1e74b1499990121d67518f4928221898d5459fb650ca52410a1b67e2
09:17:42 976f273d0409558daae8d13f1ebab1ed6add11c5e5ea8b10e76a671c8281eefa
09:17:42 b75e81287038111ea898ce1aeb8d5e5ae70ad9bdc12e1088e143be692ea01faa
09:17:42 ERROR: 3 pods left and found at /run/vc/sbs
```

This seems unrelated to this PR and may just need a cleanup on their machine.

File: .ci/configure_containerd_for_kata.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@ set -o nounset
 set -o pipefail
 
 if [ "$KATA_BUILD_CC" == "yes" ]; then
```
Comment (wainersm) [inline .ci/configure_containerd_for_kata.sh:12]: This entire `if` block can be removed as `PREFIX="${PREFIX:-/opt/kata}"`  is assigned next (line 15).

File: .ci/configure_crio_for_kata.sh [hunk 2]
```diff
@@ -12,7 +12,7 @@ set -o pipefail
 source /etc/os-release || source /usr/lib/os-release
 
 if [ "$KATA_BUILD_CC" == "yes" ]; then
```
Comment (wainersm) [inline .ci/configure_crio_for_kata.sh:14]: likewise

File: .ci/install_runtime.sh [hunk 3]
```diff
@@ -20,7 +20,7 @@ KATA_EXPERIMENTAL_FEATURES="${KATA_EXPERIMENTAL_FEATURES:-}"
 MACHINETYPE="${MACHINETYPE:-q35}"
 METRICS_CI="${METRICS_CI:-}"
 if [ "$KATA_BUILD_CC" == "yes" ]; then
```
Comment (wainersm) [inline .ci/install_runtime.sh:22]: likewise

File: .ci/install_td_shim.sh [hunk 4]
```diff
@@ -17,7 +17,7 @@ main() {
 	build_static_artifact_and_install "tdx-td-shim"
 
 	if [ "${KATA_BUILD_CC:-no}" == "yes" ]; then
-		sudo ln -sf /opt/confidential-containers/share/td-shim \
+		sudo ln -sf /opt/kata/share/td-shim \
```
Comment (wainersm) [inline .ci/install_td_shim.sh:20]: I never understood those td-shim and tdvf symlinks in `/usr/share` . My opportunity to ask: why we need them?

Other than that, could be use `$PREFIX` here to make it configurable?

File: .ci/install_runtime.sh (hunk 3 above)
Comment (fidencio) [reply .ci/install_runtime.sh:22]: Fixed.

File: .ci/install_ovmf_sev.sh [hunk 5]
```diff
@@ -19,8 +19,8 @@ main() {
 	clone_katacontainers_repo
 
 	pushd $katacontainers_repo_dir
-	sudo -E PATH=$PATH bash ${buildscript} --build=cc-sev-ovmf
```
Comment (wainersm) [inline .ci/install_ovmf_sev.sh:22]: I think I mentioned to @UnmeshDeodhar that this script should be using `build_static_artifact_and_install` like all the others. This is the kind of issue that arises when not following the "standard".

File: .ci/install_td_shim.sh (hunk 4 above)
Comment (wainersm) [reply .ci/install_td_shim.sh:20]: Yes, it's fair IMO.

Yeah, have no idea why the symlink and it seems I will not know the answer anytime soon :(

**Comment by fidencio** [comment]
> Do you know if the jobs on bare-metal are wiping out /opt/confidential-containers and /opt/kata at the end? My concern is the tests passing on those environments because we are refering artifacts installed by previous jobs.

From the TDX side, things are good.

**Review by wainersm — APPROVED** [review]
@fidencio I don't see any problem with those changes. Passing CI, we are good.

**Comment by wainersm** [comment]
> > Do you know if the jobs on bare-metal are wiping out /opt/confidential-containers and /opt/kata at the end? My concern is the tests passing on those environments because we are refering artifacts installed by previous jobs.
> 
> From the TDX side, things are good.

@BbolroC could you please validate the result of s390x?
@UnmeshDeodhar same for SEV/SNP please?

**Comment by fidencio** [comment]
TDX CI timed out while building td-shim:
```
16:41:17 Build timed out (after 20 minutes). Marking the build as aborted.
17:01:18 .ci/install_kata.sh: line 77: 1680684 Terminated              "${cidir}/install_td_shim.sh"
17:01:23 [install_kata.sh:83] ERROR: .ci/install_td_shim.sh
17:01:23 [jenkins_job_build.sh:252] ERROR: .ci/setup.sh
17:01:23 ++ handle_error 93
17:01:23 ++ local exit_code=143
17:01:23 ++ local line_number=93
17:01:23 ++ echo 'Failed at /home/jenkins/workspace/tests-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/ci_entry_point.sh +93: .ci/jenkins_job_build.sh "${repo_to_test}" -t "${TEST_TYPE}"'
17:01:23 Failed at /home/jenkins/workspace/tests-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/ci_entry_point.sh +93: .ci/jenkins_job_build.sh "${repo_to_test}" -t "${TEST_TYPE}"
17:01:23 ++ exit 143
17:01:23 Build was aborted
```

Restarting it.

**Comment by fidencio** [comment]
I hit a real issue with QEMU TDX CI, which was disabled for some time in the tests repo. :-/.

Now it's enabled again, and I'm re-running the tests.

**Comment by fidencio** [comment]
SEV tests failing again:
```
23:22:00 1..5
23:22:18 not ok 1 setup_file failed
23:22:18 # (from function `configure_cc_containerd' in file confidential/../../confidential/lib.sh, line 230,
23:22:18 #  from function `setup_file' in test file confidential/sev.bats, line 57)
23:22:18 #   `configure_cc_containerd "${saved_containerd_conf_file}"' failed with status 4
23:22:18 # Installing required packages...
23:22:18 #
23:22:18 # WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
23:22:18 #
23:22:18 # Reading package lists...
23:22:18 # Building dependency tree...
23:22:18 # Reading state information...
23:22:18 # cpuid is already the newest version (20200211-1).
23:22:18 # docker-compose is already the newest version (1.25.0-1).
23:22:18 # mysql-client is already the newest version (8.0.33-0ubuntu0.20.04.2).
23:22:18 # jq is already the newest version (1.6-1ubuntu0.20.04.1).
23:22:18 # python3-pip is already the newest version (20.0.2-5ubuntu1.9).
23:22:18 # The following packages were automatically installed and are no longer required:
23:22:18 #   libpython2-stdlib libpython2.7-minimal libpython2.7-stdlib python2
23:22:18 #   python2-minimal python2.7 python2.7-minimal
23:22:18 # Use 'sudo apt autoremove' to remove them.
23:22:18 # The following packages will be REMOVED:
23:22:18 #   python-is-python2
23:22:18 # The following NEW packages will be installed:
23:22:18 #   python-is-python3
23:22:18 # 0 upgraded, 1 newly installed, 1 to remove and 55 not upgraded.
23:22:18 # Need to get 0 B/2,364 B of archives.
23:22:18 # After this operation, 0 B of additional disk space will be used.
23:22:18 # (Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 160701 files and directories currently installed.)
23:22:18 # Removing python-is-python2 (2.7.17-4) ...
23:22:18 # Selecting previously unselected package python-is-python3.
23:22:18 # (Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 160697 files and directories currently installed.)
23:22:18 # Preparing to unpack .../python-is-python3_3.8.2-4_all.deb ...
23:22:18 # Unpacking python-is-python3 (3.8.2-4) ...
23:22:18 # Setting up python-is-python3 (3.8.2-4) ...
23:22:18 # WARNING: The directory '/home/jenkins/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
23:22:18 # Requirement already satisfied: sev-snp-measure in /home/jenkins/.local/lib/python3.8/site-packages (0.0.3)
23:22:18 # time="2023-07-24T21:22:08Z" level=fatal msg="getting status of runtime: rpc error: code = Unknown desc = server is not initialized yet"
23:22:18 # time="2023-07-24T21:22:13Z" level=fatal msg="getting status of runtime: rpc error: code = Unknown desc = server is not initialized yet"
23:22:18 # Another app is currently holding the xtables lock. Perhaps you want to use the -w option?
23:22:18 # ls: cannot access '/tmp/test-kata-sev.zssAWfaY/*.yaml': No such file or directory
23:22:18 # /home/jenkins/workspace/tests-CCv0-ubuntu-20.04_sev-x86_64-CC_SEV_CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/tests/integration/kubernetes/confidential/../../confidential/lib.sh: line 401: cd: /tmp/simple-kbs/simple-kbs: No such file or directory
23:22:18 # bats warning: Executed 1 instead of expected 5 tests
23:22:18 [run_kubernetes_tests.sh:153] ERROR: bats confidential/sev.bats
23:22:18 INFO: Run the cleanup routine
23:22:18 [cleanup_env.sh:32] INFO: Reset Kubernetes
23:22:18 [reset] Reading configuration from the cluster...
23:22:18 [reset] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -o yaml'
23:22:19 W0724 21:22:19.107460 1583364 utils.go:69] The recommended value for "resolvConf" in "KubeletConfiguration" is: /run/systemd/resolve/resolv.conf; the provided value is: /run/systemd/resolve/resolv.conf
23:22:19 [preflight] Running pre-flight checks
23:22:19 [reset] Stopping the kubelet service
23:22:19 [reset] Unmounting mounted directories in "/var/lib/kubelet"
23:22:21 [reset] Deleting contents of config directories: [/etc/kubernetes/manifests /etc/kubernetes/pki]
23:22:21 [reset] Deleting files: [/etc/kubernetes/admin.conf /etc/kubernetes/kubelet.conf /etc/kubernetes/bootstrap-kubelet.conf /etc/kubernetes/controller-manager.conf /etc/kubernetes/scheduler.conf]
23:22:21 [reset] Deleting contents of stateful directories: [/var/lib/etcd /var/lib/kubelet /var/lib/dockershim /var/run/kubernetes /var/lib/cni]
23:22:21 
23:22:21 The reset process does not clean CNI configuration. To do so, you must remove /etc/cni/net.d
23:22:21 
23:22:21 The reset process does not reset or clean up iptables rules or IPVS tables.
23:22:21 If you wish to reset iptables, you must do so manually by using the "iptables" command.
23:22:21 
23:22:21 If your cluster was setup to utilize IPVS, run ipvsadm --clear (or similar)
23:22:21 to reset your system's IPVS tables.
23:22:21 
23:22:21 The reset process does not clean your kubeconfig files and you must remove them manually.
23:22:21 Please, check the contents of the $HOME/.kube/config file.
23:22:21 [cleanup_env.sh:36] INFO: Teardown the registry server
23:22:21 [cleanup_env.sh:436] INFO: restart docker service
23:22:24 [cleanup_env.sh:92] INFO: remove local registry server: kata-registry
23:22:25 [cleanup_env.sh:40] INFO: Stop containerd service
23:22:25 [cleanup_env.sh:43] INFO: Remove network devices
23:22:25 [cleanup_env.sh:45] INFO: remove device: cni0
23:22:25 [cleanup_env.sh:45] INFO: remove device: flannel.1
23:22:25 bash: /home/jenkins/workspace/tests-CCv0-ubuntu-20.04_sev-x86_64-CC_SEV_CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/tests/integration/kubernetes/cleanup_bare_metal_env.sh false: No such file or directory
23:22:25 [cleanup_env.sh:55] INFO: Check no kata processes are left behind after reseting kubernetes
23:22:26 [cleanup_env.sh:58] INFO: Checks that pods were not left
23:22:26 Not /run/vc/sbs directory found
23:22:26 make: *** [Makefile:113: cc-sev-kubernetes] Error 1
```

**Comment by fidencio** [comment]
CLH TDX passed, but didn't update the results: http://10.112.240.228:8080/job/tests-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/569/

**Comment by UnmeshDeodhar** [comment]
> SEV tests failing again:
> 
> ```
> 23:22:00 1..5
> 23:22:18 not ok 1 setup_file failed
> 23:22:18 # (from function `configure_cc_containerd' in file confidential/../../confidential/lib.sh, line 230,
> 23:22:18 #  from function `setup_file' in test file confidential/sev.bats, line 57)
> 23:22:18 #   `configure_cc_containerd "${saved_containerd_conf_file}"' failed with status 4
> 23:22:18 # Installing required packages...
> 23:22:18 #
> 23:22:18 # WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
> 23:22:18 #
> 23:22:18 # Reading package lists...
> 23:22:18 # Building dependency tree...
> 23:22:18 # Reading state information...
> 23:22:18 # cpuid is already the newest version (20200211-1).
> 23:22:18 # docker-compose is already the newest version (1.25.0-1).
> 23:22:18 # mysql-client is already the newest version (8.0.33-0ubuntu0.20.04.2).
> 23:22:18 # jq is already the newest version (1.6-1ubuntu0.20.04.1).
> 23:22:18 # python3-pip is already the newest version (20.0.2-5ubuntu1.9).
> 23:22:18 # The following packages were automatically installed and are no longer required:
> 23:22:18 #   libpython2-stdlib libpython2.7-minimal libpython2.7-stdlib python2
> 23:22:18 #   python2-minimal python2.7 python2.7-minimal
> 23:22:18 # Use 'sudo apt autoremove' to remove them.
> 23:22:18 # The following packages will be REMOVED:
> 23:22:18 #   python-is-python2
> 23:22:18 # The following NEW packages will be installed:
> 23:22:18 #   python-is-python3
> 23:22:18 # 0 upgraded, 1 newly installed, 1 to remove and 55 not upgraded.
> 23:22:18 # Need to get 0 B/2,364 B of archives.
> 23:22:18 # After this operation, 0 B of additional disk space will be used.
> 23:22:18 # (Reading database ... 
> (Reading database ... 5%
> (Reading database ... 10%
> (Reading database ... 15%
> (Reading database ... 20%
> (Reading database ... 25%
> (Reading database ... 30%
> (Reading database ... 35%
> (Reading database ... 40%
> (Reading database ... 45%
> (Reading database ... 50%
> (Reading database ... 55%
> (Reading database ... 60%
> (Reading database ... 65%
> (Reading database ... 70%
> (Reading database ... 75%
> (Reading database ... 80%
> (Reading database ... 85%
> (Reading database ... 90%
> (Reading database ... 95%
> (Reading database ... 100%
> (Reading database ... 160701 files and directories currently installed.)
> 23:22:18 # Removing python-is-python2 (2.7.17-4) ...
> 23:22:18 # Selecting previously unselected package python-is-python3.
> 23:22:18 # (Reading database ... 
> (Reading database ... 5%
> (Reading database ... 10%
> (Reading database ... 15%
> (Reading database ... 20%
> (Reading database ... 25%
> (Reading database ... 30%
> (Reading database ... 35%
> (Reading database ... 40%
> (Reading database ... 45%
> (Reading database ... 50%
> (Reading database ... 55%
> (Reading database ... 60%
> (Reading database ... 65%
> (Reading database ... 70%
> (Reading database ... 75%
> (Reading database ... 80%
> (Reading database ... 85%
> (Reading database ... 90%
> (Reading database ... 95%
> (Reading database ... 100%
> (Reading database ... 160697 files and directories currently installed.)
> 23:22:18 # Preparing to unpack .../python-is-python3_3.8.2-4_all.deb ...
> 23:22:18 # Unpacking python-is-python3 (3.8.2-4) ...
> 23:22:18 # Setting up python-is-python3 (3.8.2-4) ...
> 23:22:18 # WARNING: The directory '/home/jenkins/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
> 23:22:18 # Requirement already satisfied: sev-snp-measure in /home/jenkins/.local/lib/python3.8/site-packages (0.0.3)
> 23:22:18 # time="2023-07-24T21:22:08Z" level=fatal msg="getting status of runtime: rpc error: code = Unknown desc = server is not initialized yet"
> 23:22:18 # time="2023-07-24T21:22:13Z" level=fatal msg="getting status of runtime: rpc error: code = Unknown desc = server is not initialized yet"
> 23:22:18 # Another app is currently holding the xtables lock. Perhaps you want to use the -w option?
> 23:22:18 # ls: cannot access '/tmp/test-kata-sev.zssAWfaY/*.yaml': No such file or directory
> 23:22:18 # /home/jenkins/workspace/tests-CCv0-ubuntu-20.04_sev-x86_64-CC_SEV_CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/tests/integration/kubernetes/confidential/../../confidential/lib.sh: line 401: cd: /tmp/simple-kbs/simple-kbs: No such file or directory
> 23:22:18 # bats warning: Executed 1 instead of expected 5 tests
> 23:22:18 [run_kubernetes_tests.sh:153] ERROR: bats confidential/sev.bats
> 23:22:18 INFO: Run the cleanup routine
> 23:22:18 [cleanup_env.sh:32] INFO: Reset Kubernetes
> 23:22:18 [reset] Reading configuration from the cluster...
> 23:22:18 [reset] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -o yaml'
> 23:22:19 W0724 21:22:19.107460 1583364 utils.go:69] The recommended value for "resolvConf" in "KubeletConfiguration" is: /run/systemd/resolve/resolv.conf; the provided value is: /run/systemd/resolve/resolv.conf
> 23:22:19 [preflight] Running pre-flight checks
> 23:22:19 [reset] Stopping the kubelet service
> 23:22:19 [reset] Unmounting mounted directories in "/var/lib/kubelet"
> 23:22:21 [reset] Deleting contents of config directories: [/etc/kubernetes/manifests /etc/kubernetes/pki]
> 23:22:21 [reset] Deleting files: [/etc/kubernetes/admin.conf /etc/kubernetes/kubelet.conf /etc/kubernetes/bootstrap-kubelet.conf /etc/kubernetes/controller-manager.conf /etc/kubernetes/scheduler.conf]
> 23:22:21 [reset] Deleting contents of stateful directories: [/var/lib/etcd /var/lib/kubelet /var/lib/dockershim /var/run/kubernetes /var/lib/cni]
> 23:22:21 
> 23:22:21 The reset process does not clean CNI configuration. To do so, you must remove /etc/cni/net.d
> 23:22:21 
> 23:22:21 The reset process does not reset or clean up iptables rules or IPVS tables.
> 23:22:21 If you wish to reset iptables, you must do so manually by using the "iptables" command.
> 23:22:21 
> 23:22:21 If your cluster was setup to utilize IPVS, run ipvsadm --clear (or similar)
> 23:22:21 to reset your system's IPVS tables.
> 23:22:21 
> 23:22:21 The reset process does not clean your kubeconfig files and you must remove them manually.
> 23:22:21 Please, check the contents of the $HOME/.kube/config file.
> 23:22:21 [cleanup_env.sh:36] INFO: Teardown the registry server
> 23:22:21 [cleanup_env.sh:436] INFO: restart docker service
> 23:22:24 [cleanup_env.sh:92] INFO: remove local registry server: kata-registry
> 23:22:25 [cleanup_env.sh:40] INFO: Stop containerd service
> 23:22:25 [cleanup_env.sh:43] INFO: Remove network devices
> 23:22:25 [cleanup_env.sh:45] INFO: remove device: cni0
> 23:22:25 [cleanup_env.sh:45] INFO: remove device: flannel.1
> 23:22:25 bash: /home/jenkins/workspace/tests-CCv0-ubuntu-20.04_sev-x86_64-CC_SEV_CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/tests/integration/kubernetes/cleanup_bare_metal_env.sh false: No such file or directory
> 23:22:25 [cleanup_env.sh:55] INFO: Check no kata processes are left behind after reseting kubernetes
> 23:22:26 [cleanup_env.sh:58] INFO: Checks that pods were not left
> 23:22:26 Not /run/vc/sbs directory found
> 23:22:26 make: *** [Makefile:113: cc-sev-kubernetes] Error 1
> ```

Based on this:
` time="2023-07-24T21:22:08Z" level=fatal msg="getting status of runtime: rpc error: code = Unknown desc = server is not initialized yet"
16:22:18 # time="2023-07-24T21:22:13Z" level=fatal msg="getting status of runtime: rpc error: code = Unknown desc = server is not initialized yet"
16:22:18 # Another app is currently holding the xtables lock. Perhaps you want to use the -w option?`

And this explanation: https://serverfault.com/questions/805718/iptables-another-app-is-currently-holding-the-xtables-lock

Looks like another networking related issue. Hopefully re-run should pass.

**Pull request merged**